static/panes/editor.js

```diff
@@ -598,6 +609,17 @@ Editor.prototype.onSettingsChange = function (newSettings) {
         this.onEditorSetDecoration(this.id, -1, false);
     }
 
+    if (before.useVim !== after.useVim) {
+        if (this.vimMode) {
```

Comment:
Nice: thanks for remembering to `dispose()`, and for taking the time to check `before` and `after` with `!==` :)